dandiapi/api/views/dashboard.py

```diff
@@ -82,6 +83,39 @@ def _users(self):
         )
 
 
+def mailchimp_csv_view(request: HttpRequest) -> StreamingHttpResponse:
+    """Generate a Mailchimp-compatible CSV file of all active users."""
+    # In production, there's a placeholder user with a blank email that we want
+    # to avoid.
+    users = User.objects.filter(is_active=True).exclude(email='')
```

Comment:
```suggestion
    users = User.objects.filter(metadata__status=UserMetadata.Status.APPROVED).exclude(email='')
```

The built-in `is_active` field is used to determine if the user can login or not. A PENDING user (i.e. a user waiting for approval) is still allowed to login, they just can't do most operations, so they are actually have `is_active=True`.